### ISSUE TITLE
Invalid Cast Exception (Imported BLPs)

### ISSUE THREAD

**Issue by Kithio** [Drake53/War3Net#34]
Only thing that's been added is a few BLPs for Icons currently, noticed this problem where it makes the map unable to be built through War3Net if an `.imp` file exists in my map source

I'm running the latest available builds from NuGet (War3Net.Build 5.5.3) alongside the CSharpLua packages, if I remove the imports file and import the files from external source and let the compiler figure out the rest it works fine, but problem is I still want to have access to this data through the Object Editor in World Editor.

I saw a similar post which mentioned a casting issue in which you said it was fixed in 5.5.4, but that version only currently exists for `War3Net.Build.Core`

```
System.InvalidCastException
  HResult=0x80004002
  Message=Unable to cast object of type 'System.Byte' to type 'War3Net.Build.Import.ImportedFileFlags'.
  Source=System.Private.CoreLib
  StackTrace:
   at System.Runtime.CompilerServices.CastHelpers.Unbox(Void* toTypeHnd, Object obj)
   at War3Net.Common.EnumConvert`1.FromByte(Byte value, Boolean allowNoFlags)
   at War3Net.Build.Import.ImportedFiles.ReadFrom(BinaryReader reader) in /_/src/War3Net.Build.Core/Import/ImportedFiles.cs:line 37
   at War3Net.Build.Import.ImportedFiles..ctor(BinaryReader reader) in /_/src/War3Net.Build.Core/Import/ImportedFiles.cs:line 26
   at War3Net.Build.Import.MapImportedFiles..ctor(BinaryReader reader) in /_/src/War3Net.Build.Core/Import/MapImportedFiles.cs:line 28
   at War3Net.Build.Extensions.BinaryReaderExtensions.ReadMapImportedFiles(BinaryReader reader) in /_/src/War3Net.Build.Core/Extensions/BinaryReaderExtensions.cs:line 73
   at War3Net.Build.Map..ctor(String mapName, String mapFolder, MapFiles mapFiles) in /_/src/War3Net.Build.Core/Map.cs:line 135
   at War3Net.Build.Map.Open(String path, MapFiles mapFiles) in /_/src/War3Net.Build.Core/Map.cs:line 467
   at Launcher.Program.Build(Boolean launch) in D:\Development\Warcraft 3\WCSharp - Red Storm\Launcher\Program.cs:line 80
   at Launcher.Program.MakeDecision() in D:\Development\Warcraft 3\WCSharp - Red Storm\Launcher\Program.cs:line 64
   at Launcher.Program.Main() in D:\Development\Warcraft 3\WCSharp - Red Storm\Launcher\Program.cs:line 43
```

```
public static TEnum FromByte(byte value, bool allowNoFlags = true) {
	TEnum val = (TEnum)(object)value; <-- Line that's throwing a fit
	
	...
}
```

**Comment by Drake53** [comment]
I didn't bother updating War3Net.Build because the fix was in War3Net.Build.Core, if you add a reference to War3Net.Build v5.5.3 and War3Net.Build.Core v5.5.4 it should work.

**Comment by YakaryBovine** [comment]
Hi @Drake53, I'm getting this error as well despite having my package references set up as follows:
```
<PackageReference Include="War3Net.Build" Version="5.5.3"/>
<PackageReference Include="War3Net.Build.Core" Version="5.5.4"/>
```

**Comment by Drake53** [comment]
Should be fixed in v5.5.5, let me know if you're still getting errors.

**Comment by YakaryBovine** [comment]
Just tested it and I am no longer getting this error. Thank you so much.
